MekHQ/src/mekhq/gui/utilities/StaticChecks.java

```diff
@@ -405,6 +405,42 @@ public static boolean doAllUnitsHaveC3Master(Vector<Unit> units) {
         return true;
 
     }
+    
+    /**
+     * Used to test a selection of Units provided by the player and determine whether they all share a designated unitType. 
+     * @param units Vector of units that the player has selected
+     * @return false if any unit in the passed-in Vector does not have the specified unit type
+     */
+    public static boolean areAllUnitsSameType(Vector<Unit> units, int unitType) {
+        // For our purposes, a selection of tanks isn't the same unless they're all the same weight class
+        // Set this based on the first unit in the selection
+        double firstUnitWeight = units.get(0).getEntity() != null ? units.get(0).getEntity().getWeight() : 0;
```

Comment:
This should exit early if `units` is empty.

Reply:
Fixed.  I thought I only called this in the first place after checking that in TOEMouseAdapter, but that is only done in one of two locations.